whois-nrtm4/src/main/java/net/ripe/db/nrtm4/util/NrtmFileUtil.java

```diff
@@ -0,0 +1,49 @@
+package net.ripe.db.nrtm4.util;
+
+import com.google.common.hash.Hashing;
+import net.ripe.db.nrtm4.publish.PublishableDeltaFile;
+import net.ripe.db.nrtm4.publish.PublishableSnapshotFile;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Random;
+import java.util.UUID;
+
+import static net.ripe.db.nrtm4.NrtmConstants.DELTA_PREFIX;
+import static net.ripe.db.nrtm4.NrtmConstants.SNAPSHOT_PREFIX;
+
+
+@Service
+public class NrtmFileUtil {
```

Comment:
there is no need for this class to be a service?

Reply:
changed to @Component.

Reply:
Sorry I meant you do not need component or service annotation for a util class

Reply:
I agree you can make all methods static and don't allow `NrtmFileUtil` to be istantiated, so there is no temptation to store any state here, e.g. see `RestClientUtils`

Reply:
i made it injectable it so that the randomness can be overridden for use in unit tests -- otherwise it can't be tested. do you think it's better to use an interface and use a different implementation for tests? i'm thinking of the session id, which is put into JSON files and hashed.

Reply:
We should avoid coding thinking about writing unit tests and I am not sure how making it injectable avoid randomness.  Lets talk separately may be I can help

Reply:
my mistake -- it's actually an integration test. changed now to be completely static.